query-languages/m/m-spec-types.md

```diff
@@ -330,9 +330,17 @@ The following are pairwise equivalent (for any <code>type <em>T</em></code>):
 
 ## Ascribed type of a value
 
-A value's _ascribed type_ is the type to which a value is _declared_ to conform. When a value is ascribed a type, only a limited conformance check occurs. _M does not perform conformance checking beyond a nullable primitive type. M program authors that choose to ascribe values with type definitions more complex than a nullable primitive-type must ensure that such values conform to these types._
+A value's _ascribed type_ is the type to which a value is _declared_ to conform. 
 
-A value may be ascribed a type using the library function `Value.ReplaceType`. The function either returns a new value with the type ascribed or raises an error if the new type is incompatible with the value's native primitive type. In particular, the function raises an error when an attempt is made to ascribe an abstract type, such as `any`.
+A value may be ascribed a type using the library function `Value.ReplaceType`. This function either returns a new value with the type ascribed or raises an error if the new type is incompatible with the value.
+
+When a value is ascribed a type, only a limited conformance check occurs:
+* The type being ascribed must be non-abstract, not nullable and compatible with the value's intrinsic (native) _primitive-type_.
+* When a custom type is being ascribed, it must match in structure with the value.
+  * For records: The type must define the same number of fields as the value. (However, field names and field types are not checked.)
+  * For tables: The type must define the same number of columns as the value. (However, column names and column types are not checked.)
+  * For functions: The type must define the same number of required parameters, as well as the same number of optional parameters, as the value. (However, parameter names, parameter types and the function return type are not checked.)
+  * For lists: As a list type always defines exactly one item type, any list type is structurally compatible with all list values.
```

Comment:
No. Trying to say that, since a list's structure is always defined by exactly and only a single item type, any two list types will always be structurally compatible, as both are intrinsically guaranteed to have a single item type. 

Is the revision I just made any clearer?

Reply:
I think I understand what you're getting at now, but it seems like trying to explain that might be more confusing than helpful.